viewer/src/datamodels/cad/sector/CachedRepository.ts

```diff
@@ -54,200 +68,183 @@ export class CachedRepository implements Repository {
     data: SectorGeometry | SectorQuads;
   }> = new Subject();
 
+  private readonly _concurrentNetworkOperations: number;
   private isDisposed = false;
 
   constructor(
     modelSectorProvider: CadSectorProvider,
     modelDataParser: CadSectorParser,
-    modelDataTransformer: SimpleAndDetailedToSector3D
+    modelDataTransformer: SimpleAndDetailedToSector3D,
+    concurrentNetworkOperations: number = 50
   ) {
     this._modelSectorProvider = modelSectorProvider;
     this._modelDataParser = modelDataParser;
     this._modelDataTransformer = modelDataTransformer;
+    this._concurrentNetworkOperations = concurrentNetworkOperations;
   }
 
-  clearSemaphore() {
-    this._rateLimiter.clearPendingRequests();
+  clearCache() {
+    this._modelDataCache.clear();
   }
 
+  getParsedData(): Observable<{ blobUrl: string; lod: string; data: SectorGeometry | SectorQuads }> {
+    return this._parsedDataSubject.pipe(distinct(keySelector => '' + keySelector.blobUrl + '.' + keySelector.sectorId)); // TODO: Should we do replay subject here instead of variable type?
+  }
   public dispose(): void {
     if (this.isDisposed) {
       return;
     }
     this.isDisposed = true;
     this._isLoadingSubject.complete();
-    this.clearSemaphore();
```

Comment:
@j-bjorne Before that thing stopped queued requests from being run after disposal. How is it handled now?

Reply:
Added some more files to handle unsubscribing. I am using the rx Subscription instead, which is made for clean up.